tests/phpunit/tests/rest-api/rest-server.php

```diff
@@ -2431,6 +2441,145 @@ public function test_rest_allowed_cors_headers_filter_receives_request_object()
 		$this->assertSame( '/test-allowed-cors-headers', $mock_hook->get_events()[0]['args'][1]->get_route() );
 	}
 
+	/**
+	 * @ticket 61739
+	 */
+	public function test_validates_request_when_building_target_hints() {
+		register_rest_route(
+			'test-ns/v1',
+			'/test/(?P<id>\d+)',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => static function () {
+						return new \WP_REST_Response();
+					},
+					'permission_callback' => '__return_true',
+					'args'                => array(
+						'id' => array(
+							'type' => 'integer',
+						),
+					),
+				),
+			)
+		);
+
+		$response = new WP_REST_Response();
+		$response->add_link( 'self', rest_url( 'test-ns/v1/test/garbage' ) );
+
+		$links = rest_get_server()::get_response_links( $response );
+
+		$this->assertArrayHasKey( 'self', $links );
+		$this->assertArrayNotHasKey( 'targetHints', $links['self'][0] );
+	}
+
+	/**
+	 * @ticket 61739
+	 */
+	public function test_sanitizes_request_when_building_target_hints() {
+		register_rest_route(
+			'test-ns/v1',
+			'/test/(?P<id>\d+)',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => static function () {
+						return new \WP_REST_Response();
+					},
+					'permission_callback' => function ( WP_REST_Request $request ) {
+						$this->assertIsInt( $request['id'] );
```

Comment:
Instead of doing the assertion here, I'd just assign it to a variable here and do the assertion later, together with the others.

This way you can ensure that it is actually executed.

Reply:
Good idea, I'll adjust that.

Reply:
How does https://github.com/WordPress/wordpress-develop/pull/7139/commits/0ffd0eba9b17a6faa25e59d8e58de7a71c21871f look?

Reply:
Exactly what I had in mind 👍